gimmik/__init__.py

```diff
@@ -29,10 +52,32 @@ def generate_mm(mat, dtype, platform, alpha=1.0, beta=0.0, funcn='gimmik_mm'):
     tpl = pkgutil.get_data(__name__, 'kernels/{0}.mako'.format(platform))
     src = Template(tpl).render(**tplargs)
 
-    # At single precision suffix all floating point constants by 'f'
-    if dtype == 'float':
-        src = re.sub(r'(?=\d*[.eE])(?=\.?\d)\d*\.?\d*(?:[eE][+-]?\d+)?',
-                     r'\g<0>f', src)
+    # Append suffix to handle typing
+    src = re.sub(r'(?=\d*[.eE])(?=\.?\d)\d*\.?\d*(?:[eE][+-]?\d+)?',
+                 rf'\g<0>{suffix}', src)
+
+    # Split lines to enforce line length max (needed for F90-F08 ISO)
+    if maxlen:
+        src = _line_split(maxlen,cchar[lang],src)
+    elif lang == 'f90':
+        warnings.warn('No maxlen given for F90 based kernel')
 
     # Return the source
     return src
+
+
+def _line_split(maxlen,cchar,src):
+    lines = src.splitlines()
+
+    src = ''
+    for line in lines:
+        nidnt = len(line) - len(line.lstrip(' '))
+
+        while math.ceil(len(line)/maxlen) > 1:
+            ns = max(line[:maxlen].rfind('+ '),line[:maxlen].rfind('- '))
```

Comment:
This does not seem safe to me.

Reply:
I tried some attempts to break this and it was fine, a line would never end in `- ` or `+ ` so its fine.